common/djangoapps/util/milestones_helpers.py

```diff
@@ -341,23 +332,33 @@ def add_course_content_milestone(course_id, content_id, relationship, milestone)
     """
     if not settings.FEATURES.get('MILESTONES_APP', False):
         return None
-    from milestones import api as milestones_api
     return milestones_api.add_course_content_milestone(course_id, content_id, relationship, milestone)
 
 
 def get_course_content_milestones(course_id, content_id, relationship, user_id=None):
     """
     Client API operation adapter/wrapper
+    Uses the request cache to store all of a user's
+    milestones
     """
     if not settings.FEATURES.get('MILESTONES_APP', False):
         return []
-    from milestones import api as milestones_api
-    return milestones_api.get_course_content_milestones(
-        course_id,
-        content_id,
-        relationship,
-        {'id': user_id} if user_id else None
-    )
+
+    if user_id is None:
+        return milestones_api.get_course_content_milestones(course_id, content_id, relationship)
+
+    request_cache_dict = request_cache.get_cache(REQUEST_CACHE_NAME)
+    if user_id not in request_cache_dict:
+        request_cache_dict[user_id] = {}
+
+    if relationship not in request_cache_dict[user_id]:
+        request_cache_dict[user_id]['requires'] = milestones_api.get_course_content_milestones(
```

Comment:
Should `'requires'` here be `relationship`?  Otherwise the list comprehension below could still fail.